dipy/denoise/localpca.py

```diff
@@ -242,7 +252,7 @@ def genpca(arr, sigma=None, mask=None, patch_radius=2, pca_method='eig',
 
                 if sigma is None:
                     # Random matrix theory
-                    this_var, ncomps = _pca_classifier(d, num_samples)
+                    this_var, ncomps = _pca_classifier(d, num_samples)  # NOTE: ncomps not usable as index if num_samples < num_features, but is recalculated below anyway 
```

Comment:
Generally, if the value  `ncomps` is not used here, it could be substituted by `_`.

Then, and having a look at a few lines below, the returned `ncomps` is never used, which is questionable. It is only used in the test. Also, following the comment, in the test there is a single voxel, so if that means `num_samples=1`, then I am not sure the value can be tested; alternatively, if
> the PCA is done on X with rows = voxels and columns = acquisitions.

as said in https://github.com/dipy/dipy/issues/2655#issue-1399383100, then again not sure if the wording is accurate (e.g. should B0 be used as suggested in the same issue ?).

A style comment: I would avoid adding in-line comments unless they are short, otherwise the large horizontal space hinders readability. Same goes to all in-line notes above and below.

Reply:
We cannot do PCA with num_samples=1 and I believe this is prevented elsewhere in the code. The code tests for how many components to keep based on the variance, rather than `ncomps`, I believe that `ncomps` was likely left in the code for testing reasons, and so you are probably right and we should remove it here, to make it clear that it is not used

Point taken on the comment, this was mainly to highly things for reviewers

Reply:
Done.